.github/workflows/_build_bodo_pip.yml

```diff
@@ -30,68 +34,97 @@ jobs:
-            NEW_PYTHON_DIR=$(cd $(dirname $(which python))/..; pwd)
-            echo "NEW_PYTHON_DIR: $NEW_PYTHON_DIR"
-            rm -rf $OLD_PYTHON_DIR
-            ln -s $NEW_PYTHON_DIR $OLD_PYTHON_DIR
-          CIBW_ENVIRONMENT: DISABLE_CCACHE=1 SCCACHE_BUCKET=engine-codebuild-cache SCCACHE_REGION=us-east-2 SCCACHE_S3_USE_SSL=true SCCACHE_S3_SERVER_SIDE_ENCRYPTION=true MACOSX_DEPLOYMENT_TARGET=${{ inputs.os == 'macos-14' && '12.0' || '10.15' }} BODO_VENDOR_MPICH=1
-          CIBW_ENVIRONMENT_PASS_LINUX: AWS_ACCESS_KEY_ID AWS_SECRET_ACCESS_KEY AWS_SESSION_TOKEN AWS_DEFAULT_REGION
-          CIBW_REPAIR_WHEEL_COMMAND_MACOS: 'delocate-wheel --ignore-missing-dependencies --sanitize-rpaths --exclude libmpi --exclude libpmpi --exclude libarrow --exclude libarrow_acero --exclude libarrow_dataset --exclude libarrow_flight --exclude libarrow_python --exclude libarrow_python_flight --exclude libarrow_python_parquet_encryption --exclude libarrow_substrait --exclude libparquet --require-archs {delocate_archs} -v {wheel} && python buildscripts/bodo/pip/macos/patch_libs_for_pip.py -p {wheel} && mv {wheel} {dest_dir}'
-          CIBW_REPAIR_WHEEL_COMMAND_LINUX: 'LD_LIBRARY_PATH=/opt/conda/lib:$LD_LIBRARY_PATH && auditwheel -v repair --exclude libmpi.so.12 --exclude libarrow.so.1801 --exclude libarrow_acero.so.1801 --exclude libarrow_dataset.so.1801 --exclude libarrow_python.so --exclude libparquet.so.1801 --plat manylinux_2_35_x86_64 {wheel} -w {dest_dir} && python buildscripts/bodo/pip/manylinux/patch_libs_for_pip.py -p {dest_dir}'
-        run: pipx run cibuildwheel==2.20.*
+            pixi install -v --locked -e pip-cpp
+            pixi list -e pip-cpp
+            # Global Tools
+            pixi global install sccache
+          CIBW_BEFORE_BUILD: |
+            # Delete mpi4py folder if it exists so correct version gets copied in
+            rm -r {package}/bodo/mpi4py || true
+            rm -rf {package}/build
+            rm -f {package}/mpich.whl
+          CIBW_ENVIRONMENT: >-
+            DISABLE_CCACHE=1
+            SCCACHE_BUCKET=engine-codebuild-cache
+            SCCACHE_REGION=us-east-2
+            SCCACHE_S3_USE_SSL=true
+            SCCACHE_S3_SERVER_SIDE_ENCRYPTION=true
+            MACOSX_DEPLOYMENT_TARGET=${{ inputs.os == 'macos-14' && '12.0' || '10.15' }}
+            BODO_VENDOR_MPICH=1
+            PATH=$HOME/.pixi/bin:$PATH
+            CONDA_PREFIX=$(pwd)/.pixi/envs/pip-cpp
+            SETUPTOOLS_SCM_PRETEND_VERSION=${{ inputs.bodo_version }}
+          CIBW_ENVIRONMENT_LINUX: >
+            PATH=/project/.pixi/envs/pip-cpp/bin:$HOME/.pixi/bin:$PATH
+            LD_LIBRARY_PATH=/project/.pixi/envs/pip-cpp/lib
+            CFLAGS="-isystem /project/.pixi/envs/pip-cpp/include"
+            CPPFLAGS="-isystem /project/.pixi/envs/pip-cpp/include"
+            CC=/project/.pixi/envs/pip-cpp/bin/x86_64-conda-linux-gnu-gcc
+            CXX=/project/.pixi/envs/pip-cpp/bin/x86_64-conda-linux-gnu-g++
+            DISABLE_CCACHE=1
```

Comment:
CIBW is annoying and wont combine the envs from `CIBW_ENVIRONMENT` and `CIBW_ENVIRONMENT_LINUX` on Linux, so I had to copy them here.

Plus, Pixi shell activation wasn't working inside of the docker container, so I had to manually set envs to find the compiler correctly.